src/Particular.Analyzers/DiagnosticDescriptors.cs

```diff
@@ -171,5 +171,21 @@ public static class DiagnosticDescriptors
             category: "Code",
             defaultSeverity: DiagnosticSeverity.Warning,
             isEnabledByDefault: true);
+
+        public static readonly DiagnosticDescriptor DateTimeAssignedToDateTimeOffset = new DiagnosticDescriptor(
+            id: DiagnosticIds.DateTimeAssignedToDateTimeOffset,
+            title: "Prevent a DateTime from being implicitly assigned to a DateTimeOffset",
```

Comment:
Nitpick: the other analyzers titles are worded in terms of the constraint, rather than what the analyzer does, so maybe (also taking into account my other comment about the name of the analyzer):

```suggestion
            title: "A DateTime should not be implicitly cast to a DateTimeOffset",
```

Mind you, this also suggests that the title of `MultipleCancellationTokensInATry` isn't quite right, and should be something like:

> A try block passing multiple cancellation tokens should be explained with a comment, and care should be taken to use the correct cancellation token in a catch block.